presto-main/src/main/java/com/facebook/presto/operator/NativeExecutionOperator.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.presto.common.Page;
+import com.facebook.presto.execution.ScheduledSplit;
+import com.facebook.presto.execution.TaskSource;
+import com.facebook.presto.execution.buffer.PagesSerdeFactory;
+import com.facebook.presto.execution.scheduler.TableWriteInfo;
+import com.facebook.presto.memory.context.LocalMemoryContext;
+import com.facebook.presto.spi.UpdatablePageSource;
+import com.facebook.presto.spi.page.PagesSerde;
+import com.facebook.presto.spi.plan.PlanNodeId;
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.google.common.base.Suppliers;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+/*
+ * TODO: The lifecycle of NativeExecutionOperator is:
+ *  1. Launch the native engine external process when initialing the operator.
+ *  2. Serialize and pass the planFragment, tableWriteInfo and taskSource to the external process through NativeExecutionTask APIs.
+ *  3. Blocking on NativeExecutionTask's getResult call to retrieve the data in serialized page format back external process.
```

Comment:
Blocking -> Block
serialized page format -> {@link SerializedPage}
back -> back from

No this is not true. You need to continuously call pollResult() to retrieve results from NativeExecutionTask.